test/expr/unary/keypath/keypath.swift

```diff
@@ -897,3 +897,11 @@ func testSyntaxErrors() { // expected-note{{}}
   _ = \A.a?;
   _ = \A.a!;
 } // expected-error@+1{{}}
+
+// SR-12290: Ban keypaths with contextual root and without a leading dot
+struct SR_12290 {
+  let property: [Int] = []
+  let kp1: KeyPath<SR_12290, Int> = \property.count // expected-error {{a Swift key path with contextual root must start with a leading dot}}{{38-38=.}}
+  let kp2: KeyPath<SR_12290, Int> = \.property.count // Ok
+  let kp3: KeyPath<SR_12290, Int> = \SR_12290.property.count // Ok
```

Comment:
Nit: Could you please a test-case where key path is used as a default argument as well?

Reply:
Done!